app/packs/review/_components/UnreviewedPack.tsx

```diff
@@ -8,41 +18,199 @@ import {
+export default function UnreviewedPack({
+	id,
+	name,
+	description,
+	type,
+	questions,
+	isDenied,
+	onStatusChange
+}: UnreviewedPackProps) {
+	const [rejectionReason, setRejectionReason] = useState('')
+	const [isLoading, setIsLoading] = useState(false)
+	const [dialogState, setDialogState] = useState<'accept' | 'reject' | null>(
+		null
+	)
+
+	const handleAction = async (action: 'accept' | 'reject') => {
+		setIsLoading(true)
+		try {
+			const response = await fetch('/api/packs/review', {
+				method: 'PUT',
+				headers: {
+					'Content-Type': 'application/json'
+				},
+				body: JSON.stringify({
+					id,
+					approved: action === 'accept' ? true : false,
+					message: action === 'reject' ? rejectionReason : undefined
+				})
+			})
+
+			if (!response.ok) throw new Error('Failed to process request')
+
+			toast({
+				title: action === 'accept' ? 'Pack Accepted' : 'Pack Rejected',
+				description:
+					action === 'accept' ?
+						'The pack has been approved and is now public'
+					:	'The pack has been rejected and the author will be notified'
+			})
+
+			setDialogState(null)
+			setRejectionReason('')
+			onStatusChange?.()
+		} catch (error) {
+			toast({
+				title: 'Error',
+				description: 'Failed to process the request',
+				variant: 'destructive'
+			})
+		} finally {
+			setIsLoading(false)
+		}
+	}
+
+	const handleDialogClose = () => {
+		// Reset rejection reason when dialog closes
+		setRejectionReason('')
+		setDialogState(null)
+	}
+
 	return (
-		<li className="p-[3px]">
-			<Card className="border-none h-full">
-				<CardHeader className="relative">
-					<CardTitle>{pack.name}</CardTitle>
-					<CardDescription>{pack.description}</CardDescription>
-				</CardHeader>
+		<Card className="relative border-none h-full flex flex-col justify-between">
+			<CardHeader>
+				<CardTitle>{name}</CardTitle>
+				<CardDescription>{description}</CardDescription>
+			</CardHeader>
+			<div>
 				<CardContent className="grid grid-cols-2">
 					<div className="flex flex-col gap-0.5">
 						<h3 className="text-sm text-muted-foreground">Questions</h3>
-						<p className="">{pack.questions.length}</p>
+						<p className="">{questions.length}</p>
 					</div>
 					<div className="flex flex-col gap-0.5">
 						<h3 className="text-sm text-muted-foreground">Type</h3>
-						<p className="">{pack.type}</p>
+						<p className="">{packMap[type as PackType]}</p>
 					</div>
 				</CardContent>
-				<CardFooter>
-					<UnreviewedPackDetails {...pack} />
+				<CardFooter className="grid grid-cols-2 gap-2 mt-auto h-fit">
+					<AlertDialog
+						open={dialogState === 'accept'}
+						onOpenChange={(open) => !open && handleDialogClose()}
+					>
+						<AlertDialogTrigger asChild>
+							<Button
+								variant="default"
+								className="bg-emerald-500 hover:bg-emerald-600 text-white"
+								disabled={isLoading}
+								onClick={() => setDialogState('accept')}
+							>
```

Comment:
Lets delete these custom override styles and instead make a new button variant for our "success" color
```suggestion
							<Button
								variant="success"
								disabled={isLoading}
								onClick={() => setDialogState('accept')}
							>
```